### PR TITLE
Electric Field of a Sphere

### PR DESCRIPTION
**Pull request by abruns123**
The electricSphere module uses the getField function to produce the vector which defines the electric field of a spherical shell that may be defined by the user in terms of its center relative to the origin, its charge density, and radius. The user also determines the point at which the electric field is measured. 

This is all tested by electricSphereTester.

The theory markdown file goes over how to determine the electric field of a sphere with constant volume charge density.

### CODE REVIEW

**Comment by laserlab** [comment]
@Cosmos491 please review asap

**Comment by Cosmos491** [comment]
@laserlab I am on it, professor.

**Comment by Cosmos491** [comment]
I will start from the beginning:

1. The first code calculates the electric field produced by a charged sphere. If the point lies inside the sphere, the field magnitude is calculated using a smaller effective radius.  The elecField function provides the direction and magnitude of the electric field at point p relative to the sphere center o.

2.  The second code is the unit test code for the electricSphere module.

3. The explanation concludes with the fact that: Inside the sphere, the electric field has a linear relationship with r. Outside, the electric field has a decreasing parabolic relationship with r. It also mentions that the electric field is a vector quantity and that the derived equations describe the magnitude of the electric field in the radial direction from the center.


Below I will review them in the same manner:

1. There is a potential issue with dividing by zero in getUVec(), when the magnitude is zero. So, I suggest adding error handling for potential division by zero in getUVec. Simplify the implementation of zeroCondition for better readability.

2. Simplify and improve the readability of the test_behavior function by breaking down the logic into smaller parts. Replace floating-point equality checks with assert_allclose to prevent errors due to rounding. 

3. The phrase "sphereical shell" should be corrected to "spherical shell".  Consider breaking down the explanation into smaller sections for clarity. But, overall the theory looks fine.

**Comment by abruns123** [comment]
At this time, I have made adjustments according to your specifications.

**Review by laserlab — CHANGES_REQUESTED** [review]
- lambda function use
- main loop

File: electricSphere.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+"""
+electricSphere uses information generated by the user to 
+calculate the electric field at a point in space produced 
+by a sphere with constant charge density placed at another
+point in space.
+"""
+import numpy as np
+
+v=lambda r:4/3*np.pi*r**3
```
[the remark this answers is not in the document]
Comment (abruns123) [reply electricSphere.py]: Okay, I've adjusted the implementation such that I have get_volume, get_area, etc functions and each return a lambda function defining volume, area etc. This got rid of the linting error involving the unnecessary lambda assignment.

File: electricSphere.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+def get_field(o,cd,r,p):
+    """getField just prints out the result of elecField in an easily readable way."""
+    e=elec_field(o,cd,r,p)
+    return (
+        "The electric Field is defined by the vector\
+ ["+str(e[0])+"x,", str(e[1])+"y,",str(e[2])+"z] V/m"
+    )
+
+if __name__=="__main__":
```
[the remark this answers is not in the document]
Comment (abruns123) [reply electricSphere.py:92]: The main loop allows the user to test the module with some control over origin of sphere, charge density, radius of sphere, and the point where the field is measured. I added a docstring to explain that more thoroughly.

[the remark this answers is not in the document]
Comment (laserlab) [reply electricSphere.py:92]: testing should be done in the unit test

**Comment by laserlab** [comment]
main loop is not for testing, merging anyway

**Pull request merged**